geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java

```diff
@@ -5653,7 +5667,7 @@ private void deleteFile(final OplogFile olf) {
       if (!olf.f.exists())
         return;
       assert olf.RAFClosed;
-      if (!olf.RAFClosed || olf.raf != null) {
+      if (!olf.RAFClosed && olf.raf != null) {
```

Comment:
Is the first part of the test required? "if (!olf.RAFClosed" It sits below an assert that in order to pass must be true.

Reply:
Actually, with the new code, this if statement should never be true because of the assert above.

Reply:
Good catch. I'll remove the first part.

Reply:
Other than that, it all looks good.